pkg/stash/with_redis_sentinel.go

```diff
@@ -0,0 +1,37 @@
+package stash
+
+import (
+	errs "errors"
+
+	"github.com/go-redis/redis/v7"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/storage"
+)
+
+var (
+	// ErrNoEndpoints is returned when no endpoints were provided
+	ErrNoEndpoints = errs.New("no endpoints provided")
+)
+
+// WithRedisSentinelLock returns a distributed singleflight
+// with a redis cluster that utilizes sentinel for quorum and failover
+func WithRedisSentinelLock(endpoints []string, master, password string, checker storage.Checker) (Wrapper, error) {
+	const op errors.Op = "stash.WithRedisSentinelLock"
+	// The redis client constructor does not return an error when no endpoints
+	// are provided, so we check for ourselves.
+	if len(endpoints) == 0 {
+		return nil, ErrNoEndpoints
```

Comment:
This error is marked as exported but doesn't get used anywhere. I'd rather keep it private until there's a use case to make it public :) 

Also, let's just keep our error objects uniform across the database by replacing the instantiation with `errors.E(op, "no endpoints provided")`

Reply:
Bad habit of making the error exported, I'll resolve that. I wasn't aware that `errors.E()` accepted a string though, need to RTFM.

Reply:
@twexler in case you didn't find it, `errors.E()` can take a variety of args after the `Op`, including a string. I most commonly use a string myself, and I'm pretty sure it's most widely used that way. Here's the code that shows the list of things it accepts: https://github.com/gomods/athens/blob/71c57ef7eeed364c77cb25aec89654b5518a5be4/pkg/errors/errors.go#L93-L104

Reply:
I did find it, but thanks for the pointer! I've got the change for this locally, still want to resolve the conversation about config/sentinel below.